pkg/image/api/v1/types.go

```diff
@@ -144,3 +182,44 @@ type DockerImageReference struct {
 	Tag       string
 	ID        string
 }
+
+type ImageStreamImport struct {
+	unversioned.TypeMeta `json:",inline"`
+	kapi.ObjectMeta      `json:"metadata,omitempty" description:"metadata about the image stream, name is required"`
+
+	Repository *RepositoryImport `json:"repository,omitempty" description:"if specified, import a single Docker repository's tags to this image stream"`
+	Images     []ImageImport     `json:"images,omitempty" description:"a list of images to import into this image stream"`
+}
+
+type RepositoryImport struct {
+	From kapi.ObjectReference `json:"from" description:"the source for the image repository to import; only kind DockerImage and a name of a Docker image repository is allowed"`
+	To   bool                 `json:"to,omitempty" description:"if true, the located images should be added to the image stream with the corresponding tags"`
+
+	ImportPolicy    TagImportPolicy `json:"importPolicy,omitempty" description:"policy controlling how the image is imported"`
+	IncludeManifest bool            `json:"includeManifest,omitempty" description:"if true, return the manifest for each image in the response"`
+
+	RepositoryImportStatus `json:",inline" description:"status of the attempt to import images"`
```

Comment:
> I don't think there's a strong benefit from the outcome to an end user to
> have spec /status (and this is not an object that lives on the cluster).

It's an expectation of API objects.  "I set the spec, the server gives me status".  It's the same argument for having an SAR with spec and status.  It's a consistent view of actor responsibilities